ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -378,6 +381,32 @@ short getVersion() {
     return toReturn != null ? Collections.unmodifiableNavigableSet(toReturn) : null;
   }
 
+  /**
+   * Generate bloom file by walking through all index entries in this segment and persist it.
+   * @throws IOException
+   */
+  private void generateBloomFileAndPersist() throws IOException {
+    List<IndexEntry> entries = new ArrayList<>();
+    getIndexEntriesSince(null, new FindEntriesCondition(Long.MAX_VALUE), entries, new AtomicLong(0), true);
```

Comment:
I was just looking at the code and I feel that calling into `getIndexEntriesSince()` is an overkill, mostly because those methods are meant to address a more general use case. The internal calls into `getAllValuesFromMmap()` scans both ways always, for instance. Also, more objects get created, more code gets exercised, etc. I think a simpler way to populate the bloom filter is by directly iterating over the entries:
```java
for (int i = 0; i < numberOfEntries(mmap); i++) {
  StoreKey key = getKeyAt(mmap, i);
  bloomFilter.add(ByteBuffer.wrap(key.toBytes()));
}
```

Reply:
How about make these changes in next PR? I need to take a deep look to get more contexts. Also, we can quickly fix the issue in this PR to unblock David's testing.

Reply:
That's fine, though I don't think there is much to this change.

Reply:
@jsjtzyy if you want to address it later, could you quickly create an issue to keep track? You can link this comment chain in the issue

Reply:
sure, will do it soon.